index.bs

```diff
@@ -196,6 +198,67 @@ and return {{FileSystemHandleKind/"directory"}} otherwise.
 The <dfn attribute for=FileSystemHandle>name</dfn> attribute must return the [=entry/name=] of the
 associated [=FileSystemHandle/entry=].
 
+### The {{FileSystemHandle/remove()}} method ### {#api-filesystemhandle-remove}
+
+<div class="note domintro">
+  : await |handle| . {{FileSystemHandle/remove()|remove}}()
+  : await |handle| . {{FileSystemHandle/remove()|remove}}({ {{FileSystemRemoveOptions/recursive}}: false })
+  :: Attempts to remove the entry represented by |handle| from the underlying file system.
+
+     For files or directories with multiple hardlinks or symlinks, the entry which is deleted
+     is the entry corresponding to the path which was used to obtain the handle.
+
+     Attempting to delete a file or directory that does not exist is considered success,
+     while attempting to delete a non-empty directory will result in a promise rejection.
+
+  : await |handle| . {{FileSystemHandle/remove()|remove}}({ {{FileSystemRemoveOptions/recursive}}: true })
+  :: Attempts to remove the entry represented by |handle| from the underlying file system.
+
+     If the entry is a directory, its contents will also be deleted recursively.
+
+     Attempting to delete a file or directory that does not exist is considered success.
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemHandle>remove(|options|)</dfn> method,
+when invoked, must run these steps:
+
+1. Let |result| be [=a new promise=].
+1. Run the following steps [=in parallel=]:
+  1. Let |entry| be <b>[=this=]</b>'s [=FileSystemHandle/entry=].
+  1. Let |permissionStatus| be the result of [=requesting access=]
+     given <b>[=this=]</b> and {{"readwrite"}}.
+     If that throws an exception, [=reject=] |result| with that exception and abort.
+  1. If |permissionStatus| is not {{PermissionState/"granted"}},
+     [=/reject=] |result| with a {{NotAllowedError}} and abort.
+
```

Comment:
We need to acquire an exclusive lock here (and also in `dir.removeEntry(file)`. I'm waiting for the language in https://github.com/whatwg/fs/pull/21 to mature so we can copy/paste the "grab an exclusive lock or fail" algorithm steps here.

Just a heads up that currently, we have [a WPT](https://github.com/web-platform-tests/wpt/blob/335c36d606c02daa4a35bd073d8c0bbd96a2fad3/fs/script-tests/FileSystemDirectoryHandle-removeEntry.js#L90-L103) (which was written before file locking was a thing) asserting that removal (via `removeEntry()`) succeeds when the file has an open writable. Since creating a writable grabs a shared lock, we'll have to update this WPT to make removal fail while any locks on the handle are held

Reply:
This WPT has [since been updated](https://github.com/web-platform-tests/wpt/blob/2ac776fb32d8f4cfe08a41773dd4adfc6750baf6/fs/script-tests/FileSystemDirectoryHandle-removeEntry.js#L90-L103). Filed #138 to track updating `removeEntry()`